src/pages/workspaces/workspace/analysis/modals/CloudEnvironmentModal.js

```diff
@@ -377,7 +377,7 @@ export const CloudEnvironmentModal = ({
         ]),
         // Cloud environment button
         div({ style: toolButtonDivStyles }, [
-          isSettingsSupported(toolLabel, cloudProvider, workspace.accessLevel, workspace?.workspace?.createdDate) && h(Clickable, {
+          isSettingsSupported(toolLabel, cloudProvider, workspace?.accessLevel, workspace?.workspace?.createdDate) && h(Clickable, {
```

Comment:
Is it possible that workspace will not be defined here?

Reply:
Looking at the other code in this file more closely, it seems the CloudEnvironmentModal does expect the `workspace` variable to be defined. I guess there might not be any need for `?`.

Reply:
Great, it would be better to not use "?" then because the code more accurately represents the expectations, and if for some reason `workspace` is not defined, the failure would be easier to diagnose.

Once this file has its variables typed it should be easier to reason about this.

Reply:
Sounds good. I will remove `?` 👍